src/pinecone.ts

```diff
@@ -219,7 +200,17 @@ export class Pinecone {
    * @returns A promise that resolves to {@link IndexMeta}
    */
   describeIndex(indexName: IndexName) {
-    return this._describeIndex(indexName);
+    const describeIndexPromise = this._describeIndex(indexName);
```

Comment:
Swapped from passing a callback to appending a `.then()` on the promise before returning.